source/usage-examples/struct-tagging.txt

```diff
@@ -0,0 +1,67 @@
+==========================
+Set Custom Driver Behavior
+==========================
+
+.. default-domain:: mongodb
+
+You can customize the way that the Go Driver converts between Go
+structs and :manual:`BSON </reference/bson-types/>` using 
+**struct tags**, which are optional instructions you can use with struct
+fields.
+
+The following code declares a ``BlogPost`` struct, where the ``WordCount`` field
+has a struct tag that sets a custom BSON field name of ``word_count``:
+
+.. literalinclude:: /includes/usage-examples/code-snippets/struct-tag.go
+   :start-after: begin struct
+   :end-before: end struct
+   :language: go
+   :dedent:
+
+The following code creates a ``BlogPost`` instance and inserts it
+into the ``posts`` collection of the ``sample_training`` database,
+converting the struct to BSON in the process:
+
+.. include:: /includes/usage-examples/run-example-tip.rst
+
+.. literalinclude:: /includes/usage-examples/code-snippets/struct-tag.go
+   :start-after: begin create and insert
+   :end-before: end create and insert
+   :language: go
+   :dedent:
+
+Click here **<TODO: github link to file>** to see a fully runnable example.
+
+Expected Result
+---------------
+
+After running the preceding code snippet, you should be able to find
+the following document in the ``posts`` collection:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+       ... ,
+       "title" : "Annuals vs. Perennials?",
+       "author" : "Sam Lee",
+       "word_count" : 682,
+       "tags" : ["seasons", "gardening", "flower"]
+   }
```

Comment:
pardon my go ignorance, but does the driver always convert init caps field names in structs to all lowercase? If not mentioned elsewhere, consider mentioning it here.